src/cli/cli.rs

```diff
@@ -269,10 +269,16 @@ impl ThinCli {
 
         // Determine the subcommand that was run while ignoring errors from yet-to-be-defined
         // arguments.
-        let matches = command
-            .clone()
-            .ignore_errors(true)
-            .get_matches_from(args.iter());
+        let matches = match cfg!(test) {
+            true => command
+                .clone()
+                .ignore_errors(true)
+                .try_get_matches_from(args.iter())?,
+            false => command
+                .clone()
+                .ignore_errors(true)
+                .get_matches_from(args.iter()),
+        };
```

Comment:
> We don't want this function to return an error in case of --help or --version, so I don't think we should special case the test version to do so. Two options IMHO:

Why not? The calls to clap's matching engine further down this function rightfully return `ClapError` instances explaining why the arguments are wrong. This helps us match against them in unit tests.

Reply:
> Right. My point is not really about what the function returns, but about it behaving differently between test and non-test builds. as the `cfg!(test)` conditional is not about some function inputs. Maybe I was not super clear, I'm not suggesting the function to never return an error, I was just speaking about the help/version error in the specific part that was updated in this PR.
> 
> I actually would prefer the function to return the correct state instead of an error in the test build and no error in the normal build; than not to return anything in case of version/help.
> 

Clap does a way better job printing the error (e.g: determining whether it should go to stdout of stderr). That's why I originally let clap do it (via calls to `ClapError::exit()`).

> You can understand the two above suggestions as,
> 
>     * Making the function not to return an error for help/version; in such case the CI has to take it into account.
> 
>     * Or, making the function returning the information ok/error/version&help.
> 

It already kind of does, you can check the kind of clap error to know if it's argument error, parsing error, or version&help. Besides, doing a mapping would not work because there is is significant private data that clap uses to print nice errors, so we'd need to end up wrapping the ClapError.


> 
> Both will help in testing the same code as what runs in the tool. IMHO this is better than making the function return an error just for CI to check if the help/version is taken into account.

We can try to refactor some `cfg` statements away of this func.